common/src/main/java/org/openmrs/analytics/FhirStoreUtil.java

```diff
@@ -56,7 +69,13 @@
 	
 	private String gcpFhirStore;
 	
-	FhirStoreUtil(String gcpFhirStore) throws IllegalArgumentException {
+	FhirStoreUtil(String gcpFhirStore, String sourceFhirUrl, String sourceUser, String sourcePw)
+	        throws IllegalArgumentException {
+		this.targetFhirStoreUrl = gcpFhirStore;
```

Comment:
I thought one goal of this change is to be able to push data into any FHIR store (not just GCP), is this correct? If yes, then why do you still match it against `FHIR_PATTERN` below? BTW, this seems to be independent of BasicAuth change, right? If yes, then please either move this to a separate PR or if you want to keep the PR this way, please also add a note about this in the PR description.

Reply:
This is another change that I didn't catch extracting the BasicAuth work from using a HAPI JPA server as the target. I'll take it out of this PR and add it to one dealing with adding additional target FhirStore options, with GCP still being one option.